### PR TITLE
Complete API routes and writing tests. Optimized DB with indexing

### PR DESCRIPTION
**Pull request by jenniferjinyoungyang**

### CODE REVIEW

File: tests/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    request.get('/products/1/related')
+      .expect(200)
+      .expect((res) => {
+        console.log(res.text);
+        expect(res.text).toBe(sampleData.relatedProducts);
+      })
+      .end(done);
+  });
+
+  // it('test get request to the endpoint /cart', (done) => {
```
Comment (tbrinkman3) [inline tests/integration.test.js]: are you still going to use these tests?

File: tests/k6tests/stressTest.productInfo.vus1.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  'vus': 1,
+  'duration': '10s',
+  'thresholds': {
+    'http_req_failed': ['rate<0.01'], // http errors should be less than 1%
+    'http_req_duration': ['p(95)<200'] // 95% of requests should be below 200ms
+  }
+};
+
+export default function () {
+  let resProductInfo = http.get('http://localhost:3001/products/1/');
```
Comment (tbrinkman3) [inline tests/k6tests/stressTest.productInfo.vus1.js:14]: for all of your stress tests, you could also include a check to make sure you are receiving the expected data, not just a response

**Review by tbrinkman3 — DISMISSED** [review]
Looks really good, I need to take a notes on from you on how to submit cleaner code before a pull request!

File: tests/integration.test.js (hunk 1 above)
Comment (jenniferjinyoungyang) [reply tests/integration.test.js]: No I'm not. I should delete this.

File: tests/k6tests/stressTest.productInfo.vus1.js (hunk 2 above)
Comment (jenniferjinyoungyang) [reply tests/k6tests/stressTest.productInfo.vus1.js:14]: Thanks for the tip. I will try that.

**Comment by jenniferjinyoungyang** [comment]
Thanks for the quick turnaround!

**Pull request merged**